lib/services/account-api-service.js

```diff
@@ -189,5 +199,107 @@ function accountApiServiceFactory(
         });
     };
 
+    AccountApiService.prototype.createRole = function (newrole) {
+        var self = this;
+        var acl = this.acl;
+        return Promise.try(function () {
+            assert.string(newrole.role);
+            return waterline.roles.findOne({ role: newrole.role });
+        })
+        .then(function (entry) {
+            if (entry) {
+                // TODO: Replace with HttpError after errors are refactored
+                var err = new Errors.BaseError('role already exists');
+                err.status = 409;
+                throw err;
+            }
+            return waterline.roles.create({
+                role: newrole.role,
+                privileges: newrole.privileges
+            });
+        })
+        .then(function (entry) {
+            return acl.addRoleParents(entry.role, entry.privileges);
+        })
+        .then(function () {
+            return self.getRoleByName(newrole.role);
+        });
```

Comment:
@keedya We may want to add a `.catch` here so we respond with a 409 in the event of a duplicate key error.  Otherwise, I think we'll return a 500 if two REST clients attempt to add a role concurrently.  In fact, we might not need the initial `findOne` query, we might be able to rely on the error thrown when a Mongo index violation occurs.

Maybe something like:
```
.catch(function(err) {
    if (err.message.match(/E11000 duplicate key error index:/) {
        var err = new Errors.BaseError('role already exists');
        err.status = 409;
        throw err;
    }
    throw err;
})
```